filtered_conn.go

```diff
@@ -0,0 +1,34 @@
+package libp2pquic
+
+import (
+	"net"
+
+	filter "github.com/libp2p/go-maddr-filter"
+)
+
+type filteredConn struct {
+	net.PacketConn
+
+	filters *filter.Filters
+}
+
+func newFilteredConn(c net.PacketConn, filters *filter.Filters) net.PacketConn {
+	return &filteredConn{PacketConn: c, filters: filters}
+}
+
+func (c *filteredConn) ReadFrom(b []byte) (n int, addr net.Addr, rerr error) {
+	for {
+		n, addr, rerr = c.PacketConn.ReadFrom(b)
+		// Short Header packet, see https://tools.ietf.org/html/draft-ietf-quic-invariants-07#section-4.2.
+		if n < 1 || b[0]&0x80 == 0 {
+			return
```

Comment:
Can we ever read a partial packet and would that cause an issue?

Reply:
In principle, yes. If the UDP packet is longer than `b`, `ReadFrom` will only read the packet up to `len(b)`, the rest of the packet will be discarded. Since every QUIC packet is integrity protected, this doesn't lead to any problems, as the partial packet will simply end up undecryptable.
QUIC itself deals with the problem of fixed buffer sizes by communication a `max_packet_size` during the handshake, so the endpoint knows not to pack any packet that is larger than its peer's buffer.